Documentation/planning/arcade-powered-source-build/intermediate-nupkg.md

```diff
@@ -35,14 +35,90 @@ source-built intermediates.
 
 ## Too large?
 
-If the artifacts are too large for our NuGet feeds to handle, we can:
-
-1. Split the monster nupkg along some logical line, and produce multiple for a
-   single repo.
-2. Split some artifacts to a tarball that is published to an auxiliary feed.
-   * Extra infra development necessary: secure transport of this tarball,
-     implementing caching, etc.
-   * Not recommended.
+Some repositories produce artifacts that would be too large for Azure DevOps
+feeds to handle if put into a single intermediate package. To solve this, some
+repos need to produce "supplemental intermediate nupkgs".
+
+A "supplemental intermediate nupkg" uses the same naming convention as
+intermediate nupkgs, but inserts an extra name between the repo name and the
+RID.
+
+* `Microsoft.SourceBuild.Intermediate.runtime.libraries.linux.x64/6.0.0-foo`
+  * Contains the dotnet/runtime platform extensions libraries.
+* `Microsoft.SourceBuild.Intermediate.runtime.Microsoft.NETCore.App.Crossgen2.linux.x64/6.0.0-foo`
+  * Contains the crossgen2 pack nupkg.
+  * `Microsoft.NETCore.App.Crossgen2` is the name of the crossgen2 pack without
+    its RID, so use that as the name by convention.
+* `Microsoft.SourceBuild.Intermediate.runtime.Microsoft.NETCore.App.Crossgen2.archive.linux.x64/6.0.0-foo`
+  * This stores the `tar.gz` version of the crossgen2 pack.
+  * Crossgen2 is huge (~167 MB), so we need one supplemental nupkg for each
+    artifact--the nupkg and the archive.
```

Comment:
When will the archive be used vs. the nupkg?  When will both be produced instead of just the nupkg?

Reply:
They're always both produced--this is talking about the difference between e.g. `dotnet-runtime-6.0.0-ci-centos.7-x64.tar.gz` and `Microsoft.NETCore.App.Runtime.centos.7-x64.6.0.0-ci.nupkg`. On the usage side, I haven't quite kept track, but I think they're both used for various reasons when piecing together the SDK. The nupkg version might be more likely to be used when the SDK targets are involved, because officially, the only way to get these is from nuget.org.

(I'm going to add the artifact filenames here in case there was just a missing connection.)